index.html

```diff
@@ -2812,6 +2812,15 @@ <h2>Algorithms</h2>
 [=map=] with a `proof` [=map/key=].
       </p>
 
+      <p>
+In addition to the algorithms below, implementers MAY implement additional
+reasonable defaults and safeguards to mitigate things such as developer error,
+excessive resource consumption, or attack models that they would like to protect
+against. The algorithms provided are the minimum requirements for an
+interoperable implementation and developers are urged to include any additional
+measures that contribute to a safer and more efficient ecosystem.
```

Comment:
Further massage may be appropriate.
```suggestion
Implementers MAY implement reasonable defaults and safeguards in addition to
the algorithms below, to mitigate things such as developer error, excessive
resource consumption, and/or attack models against which there is a particular
protection. The algorithms provided below are the minimum requirements for an
interoperable implementation, and developers are urged to include additional
measures that could contribute to a safer and more efficient ecosystem.
```

Reply:
@TallTed, well, this still reads to me as:

"to help mitigate ... attack models".

But one does not mitigate attack models (I don't think). They mitigate attacks (as described in those models). This is why I suggested language more like "... as mitigations for ...". That way it would read as "Implementers MAY implement X as mitigations":

* for developer error
* for excessive resource consumption
* for newly discovered attack models

I wouldn't die on this grammar hill though, I'll accept either your suggestion or your suggestion plus this small tweak.